### PR TITLE
Add 1.22.17 patch release to the calendar

### PR DESCRIPTION
**Pull request by xmudrii**
Follow-up from https://github.com/kubernetes/website/pull/37891#discussion_r1021413609

This PR:

- Moves 1.22 from actively supported releases to EOL releases (`Detailed Release History for Active Branches`)
- Adds 1.22.16 and 1.22.17 as dedicated/post-EOL releases with descriptions about why we released those two patch releases after the EOL date

cc @kubernetes/release-engineering @kubernetes/release-managers @kubernetes/sig-k8s-infra-leads @kubernetes/k8s-infra-gcp-org-admins

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold
for discussion

File: data/releases/schedule.yaml [hunk 1]
```diff
@@ -114,12 +114,15 @@ schedules:
...
   next:
-    release: 1.22.16
-    cherryPickDeadline: 2022-10-07
-    targetDate: 2022-10-12
+    release: 1.22.17
+    cherryPickDeadline: 2022-12-02
+    targetDate: 2022-12-07
     note: >-
-      The 1.22 release is in maintenance mode. As per the support policy, 1.22.16 will be released **only** if there are critical and/or security issues.
+      As per our support policy, the 1.22 release has reached End-of-Life (EOL), but we are planning to cut one more 1.22 patch release to migrate to the new `registry.k8s.io` image registry. Other changes will not be permitted.
```
Comment (sftim) [inline data/releases/schedule.yaml:121]: After that release is cut, change "we are planning" to reflect the new tense of the situation (ie, that it has happened) - or make some other change to cover this.

Would we still release 1.22.18 if there is a critical security issue?

Comment (cpanato) [reply data/releases/schedule.yaml:121]: maybe not, i would say only if is related to the registry topic. otherwise after this patch the 1.22 will be EOL

**Review by cpanato — APPROVED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

File: data/releases/eol.yaml [hunk 2]
```diff
@@ -1,4 +1,17 @@
 branches:
+  - release: "1.22"
+    finalPatchRelease: "1.22.17"
+    endOfLifeDate: 2022-12-08
+    note: >-
+      1.22.17 will be released in December 2022 (after the EOL date) to backport registry changes and fix two critical issues.
```
Comment (xmudrii) [inline data/releases/eol.yaml:6]: I'll make sure to change this to past tense once 1.22.17 is out.

**Pull request merged**